python-avd/pyavd/_schema/avddataconverter.py

```diff
@@ -162,28 +162,41 @@ def convert_types(self, convert_types: list, data: dict | list, index: str | int
         schema_type = schema.get("type")
 
         # Get value from input data
-        value = data[index]
+        # Pyright is not able to understand that if index is a str, data is a dict.
+        # So we have to cast it.
+        value = cast("dict", data)[index] if isinstance(index, str) else cast("list", data)[index]
 
         # For simple conversions, skip conversion if the value is of the correct type
         # Avoid corner case where we want to convert bool to int. Bool is a subclass of Int so it passes the check above.
+        if schema_type is None:
+            return
+
+        py_schema_type = SCHEMA_TO_PY_TYPE_MAP.get(schema_type)
         if (
             schema_type in SIMPLE_CONVERTERS
-            and isinstance(value, SCHEMA_TO_PY_TYPE_MAP.get(schema_type))
+            and py_schema_type is not None
+            and isinstance(value, py_schema_type)
             and not (schema_type == "int" and isinstance(value, bool))
         ):
             return
 
         for convert_type in convert_types:
-            if isinstance(value, SCHEMA_TO_PY_TYPE_MAP.get(convert_type)) and schema_type in SIMPLE_CONVERTERS:
+            py_convert_type = SCHEMA_TO_PY_TYPE_MAP.get(convert_type)
+            if py_convert_type is not None and isinstance(value, py_convert_type) and schema_type in SIMPLE_CONVERTERS:
```

Comment:
why do you need to check for the None separately?

Reply:
`.get` returns `None` as well. but I am directly fetching now.